src/Pact/Analyze/Eval/Term.hs

```diff
@@ -576,14 +576,16 @@ evalTerm = \case
     succeeds %= (.&& whetherInPact)
     view aeTrivialGuard
 
-  MkUserGuard _ty _obj _name ->
-    --
-    -- NOTE: for now we just leave this uninterpreted, until we are prepared to
-    -- close over and analyze user guard functions in the future. we do not
-    -- try to resolve two uses of the same name to the same user guard, because
-    -- the guard created is *also* a function of the supplied object.
-    --
-    genUninterpreted
+  MkUserGuard guard bodyET -> do
+    -- NOTE: a user guard can not access user tables, so we can run it ahead of
```

Comment:
is it definitely not possible for system tables to change during execution? eg a system table for gas remaining.

Reply:
My understanding is that any data from `chain-data` and `env-chain-data` would be frozen across the tx, and there is no way generally to read data that's changed since the tx started besides user tables. I could be mistaken here, though.